moto/s3/models.py

```diff
@@ -971,13 +970,17 @@ def copy_key(self, src_bucket_name, src_key_name, dest_bucket_name,
         dest_bucket = self.get_bucket(dest_bucket_name)
         key = self.get_key(src_bucket_name, src_key_name,
                            version_id=src_version_id)
-        if dest_key_name != src_key_name:
-            key = key.copy(dest_key_name)
+
+        key = key.copy(dest_key_name)
```

Comment:
theoretically this could do a check to see if the bucket was versioned, but I think this should still work ok